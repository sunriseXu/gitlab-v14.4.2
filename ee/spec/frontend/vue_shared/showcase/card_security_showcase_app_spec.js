import { shallowMount } from '@vue/test-utils';
import CardSecurityShowcaseApp from 'ee/vue_shared/showcase/card_security_showcase_app.vue';

describe('Card Security Showcase App', () => {
  let wrapper;

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  const createWrapper = () => {
    wrapper = shallowMount(CardSecurityShowcaseApp, {
      provide: {
        billingVulnerabilityManagement: '/billing?g=1',
        billingDependencyScanning: '/billing?g=2',
        billingDast: '/billing?g=3',
        billingContainerScanning: '/billing?g=4',
        trialVulnerabilityManagement: '/trial?g=1',
        trialDependencyScanning: '/trial?g=2',
        trialDast: '/trial?g=3',
        trialContainerScanning: '/trial?g=4',
      },
    });
  };

  it('renders correctly', () => {
    createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });
});
