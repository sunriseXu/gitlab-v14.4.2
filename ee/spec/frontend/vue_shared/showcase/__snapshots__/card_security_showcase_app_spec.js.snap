// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Security Showcase App renders correctly 1`] = `
<div
  class="gl-my-5"
>
  <div
    class="gl-mb-4 col-lg-6 col-xs-12 gl-p-0"
  >
    <h2
      class="gl-col-12 gl-font-size-h2 gl-text-gray-900"
    >
      
      Identify vulnerabilities in your code now
    
    </h2>
     
    <p
      class="gl-col-12 gl-text-gray-800"
    >
      
      Use GitLab CI/CD to analyze your source code for known vulnerabilities. Compare the found vulnerabilities between your source and target branches.
    
    </p>
  </div>
   
  <div
    class="gl-display-flex gl-w-full gl-flex-flow-row-wrap"
  >
    <card-showcase-stub
      description="Scan your source code using GitLab CI/CD and uncover vulnerabilities before deploying."
      primaryaction="Enable SAST"
      primarylink="/help/user/application_security/sast/index.md#configuration"
      primarylinkicon="external-link"
      primarytrackinglabel="enable_sast"
      primarytrackingproperty=""
      secondaryaction=""
      secondarylink=""
      secondarylinkicon=""
      secondarytrackinglabel=""
      secondarytrackingproperty=""
      title="Enable Static Application Security Testing (SAST)"
    />
     
    <card-showcase-stub
      description="Scan your code to detect unintentionally committed secrets, like keys, passwords, and API tokens."
      primaryaction="Enable Secret Detection"
      primarylink="/help/user/application_security/secret_detection/index.md#configuration"
      primarylinkicon="external-link"
      primarytrackinglabel="enable_secret_detection"
      primarytrackingproperty=""
      secondaryaction=""
      secondarylink=""
      secondarylinkicon=""
      secondarytrackinglabel=""
      secondarytrackingproperty=""
      title="Enable Secret Detection"
    />
  </div>
   
  <div
    class="gl-mb-4 col-lg-6 col-xs-12 gl-p-0"
  >
    <h2
      class="gl-col-12 gl-font-size-h2 gl-text-gray-900"
    >
      
      Take your security to the next level
    
    </h2>
     
    <p
      class="gl-col-12 gl-text-gray-800"
    >
      
      Start a free 30-day Ultimate trial or upgrade your instance to access organization-wide security and compliance features. See the other features of the Ultimate plan.
    
    </p>
  </div>
   
  <div
    class="gl-display-flex gl-w-full gl-flex-flow-row-wrap"
  >
    <card-showcase-stub
      description="Access a dedicated area for vulnerability management. This includes a security dashboard, vulnerability report, and settings."
      primaryaction="Start a free trial"
      primarylink="/trial?g=1"
      primarylinkicon=""
      primarytrackinglabel="start_trial"
      primarytrackingproperty="vulnerability_management"
      secondaryaction="Upgrade now"
      secondarylink="/billing?g=1"
      secondarylinkicon=""
      secondarytrackinglabel="upgrade_now"
      secondarytrackingproperty="vulnerability_management"
      title="Vulnerability management"
    />
     
    <card-showcase-stub
      description="Find out if your external libraries are safe. Run dependency scanning jobs that check for known vulnerabilities in your external libraries."
      primaryaction="Start a free trial"
      primarylink="/trial?g=2"
      primarylinkicon=""
      primarytrackinglabel="start_trial"
      primarytrackingproperty="dependency_scanning"
      secondaryaction="Upgrade now"
      secondarylink="/billing?g=2"
      secondarylinkicon=""
      secondarytrackinglabel="upgrade_now"
      secondarytrackingproperty="dependency_scanning"
      title="Dependency scanning"
    />
     
    <card-showcase-stub
      description="Dynamically examine your application for vulnerabilities in deployed environments."
      primaryaction="Start a free trial"
      primarylink="/trial?g=3"
      primarylinkicon=""
      primarytrackinglabel="start_trial"
      primarytrackingproperty="dast"
      secondaryaction="Upgrade now"
      secondarylink="/billing?g=3"
      secondarylinkicon=""
      secondarytrackinglabel="upgrade_now"
      secondarytrackingproperty="dast"
      title="Dynamic Application Security Testing (DAST)"
    />
     
    <card-showcase-stub
      description="Audit your Docker-based app. Scan for known vulnerabilities in the Docker images where your code is shipped."
      primaryaction="Start a free trial"
      primarylink="/trial?g=4"
      primarylinkicon=""
      primarytrackinglabel="start_trial"
      primarytrackingproperty="container_scanning"
      secondaryaction="Upgrade now"
      secondarylink="/billing?g=4"
      secondarylinkicon=""
      secondarytrackinglabel="upgrade_now"
      secondarytrackingproperty="container_scanning"
      title="Container scanning"
    />
  </div>
</div>
`;
